data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/LogstashPluginMapper.java

```diff
@@ -33,7 +36,10 @@ public LogstashPluginMapper() {
         this.attributesMapperProvider = attributesMapperProvider;
     }
 
-    public PluginModel mapPlugin(LogstashPlugin logstashPlugin) {
+    public List<PluginModel> mapPlugin(LogstashPlugin logstashPlugin) {
+        if(Objects.equals(logstashPlugin.getPluginName(), "mutate")) {
```

Comment:
We've added some support for more dynamic mappings using configuration files. Will that work in this instance? I do think it would be good to extend the dynamic mapping to allow it to handle this use-case rather than change code.

Reply:
And after all of this is done, the `MutateMapper` can be made to implement `LogstashCustomPluginMapper`. Then you create the mapping file.

Reply:
Can you show me more of what you are referring to?